roomserver/internal/perform/perform_join.go

```diff
@@ -271,6 +271,25 @@ func (r *Joiner) performJoinRoomByID(
 		}
 	}
 
+	// If a guest is trying to join a room, check that the room has a m.room.guest_access event
+	if req.IsGuest {
+		var guestAccessEvent *gomatrixserverlib.HeaderedEvent
+		guestAccess := "forbidden"
+		guestAccessEvent, err = r.DB.GetStateEvent(ctx, req.RoomIDOrAlias, gomatrixserverlib.MRoomGuestAccess, "")
```

Comment:
Looks like this `err` is dropped?

Reply:
The error is ignored because `guestAccess` is set to forbidden above. We only use the result of `GetStateEvent` if there's no error and we actually have a `guestAccessEvent`.

Reply:
The error could potentially be a failure contacting the database, so at the very least that should make it into the logs. Agreed we can ignore things like `sql.ErrNoRows` etc though.